### PR TITLE
monitoring: zoekt: fix heatmap configuration

### PR DESCRIPTION
**Pull request by ggilmore**
@stefanhengl and I realized that the heat maps on the Zoekt dashboards were broken. They didn't include data-points (like lengthy indexing durations) that we knew existed (we could see them in the logs).  

It turns out that we weren't using the right Grafana incantations to inform it that the data source that we're using for the heatmap is already bucketed - it doesn't need to recreate its own buckets. We fixed that configuration, and also changed the heatmap to use a higher-contrast color scheme. 

Before:

Screen Shot 2022-04-21 at 11 51 23 AM

After:

Screen Shot 2022-04-21 at 11 50 33 AM

See [this blog post](https://grafana.com/blog/2020/06/23/how-to-visualize-prometheus-histograms-in-grafana/) for the guide that we followed. 

## Test plan

We used the local monitoring stack with sourcegraph.com's Prometheus instance to see that these changes fixed the issue.

### CODE REVIEW

File: monitoring/definitions/zoekt.go [hunk 1]
```diff
@@ -686,3 +680,24 @@ func Zoekt() *monitoring.Container {
...
+		}
+	}
+
+	p.HeatmapPanel.YAxis.Format = string(monitoring.Seconds)
+	p.HeatmapPanel.YBucketBound = "upper"
+
+	p.HideZeroBuckets = true
+	p.Color.Mode = "spectrum"
+	p.Color.ColorScheme = "interpolateSpectral"
+}
```
Comment (bobheadxi) [inline monitoring/definitions/zoekt.go:703]: can we just share this as a `TimeseriesBucketsHeatmap` or similar in https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/monitoring/monitoring/panel_options.go?L41%3A1-44%3A37= ?

Comment (bobheadxi) [reply monitoring/definitions/zoekt.go:703]: For example this fix looks quite similar to what is used in https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/monitoring/definitions/shared/standard.go?L75-80

Comment (bobheadxi) [reply monitoring/definitions/zoekt.go:703]: cc @Strum355

**Review by keegancsmith — APPROVED** [review]
approving to unblock. But sharing code as Robert points outs out sounds good, but if is more effort than it looks can be done as a followup?

**Comment by ggilmore** [comment]
Yes, I can follow up on that next week.

**Pull request merged**